databricks/koalas/internal.py

```diff
@@ -638,7 +642,26 @@ def spark_df(self) -> spark.DataFrame:
     def pandas_df(self):
         """ Return as pandas DataFrame. """
         sdf = self.spark_internal_df
-        pdf = sdf.toPandas()
+
+        if LooseVersion(pyarrow.__version__) >= LooseVersion("0.15") and \
+                LooseVersion(pyspark.__version__) < LooseVersion("3.0"):
+            # There seems a bug in PyArrow 0.15 with PySpark 2.4 and below.
+            # TODO(Hyukjin): I have no idea why only inserting limit works.
+            #   Other APIs like noop map, explicitly shuffle and sample with
+            #   fraction 1 still don't work.
+            max_count = 2147483646  # JVM integer's max value - 1.
```

Comment:
BTW, if we guess each row is, say, at least 8 bytes or say 16 bytes, then the output is at least 16 GB or 32 GB. I don't think this is not quite common.